cardano-prelude/src/Data/Sequence/Strict.hs

```diff
@@ -312,6 +320,22 @@ findIndicesL p (StrictSeq xs) = Seq.findIndicesL p xs
 findIndicesR :: (a -> Bool) -> StrictSeq a -> [Int]
 findIndicesR p (StrictSeq xs) = Seq.findIndicesR p xs
 
+{-------------------------------------------------------------------------------
+  Zips and Unzips
+-------------------------------------------------------------------------------}
+
+zip :: StrictSeq a -> StrictSeq b -> StrictSeq (a, b)
+zip = zipWith (,)
```

Comment:
Even though tuples are lazy, we already know all `a`s and `b`s are in WHNF because they are in a `StrictSeq`. The `toStrict` in `zipWith` will make sure the tuples are in WHNF (this does not force the elements, just the construction of the tuple), so there will be no thunks in the `StrictSeq`.